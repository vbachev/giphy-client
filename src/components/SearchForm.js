import React from 'react'

const SearchForm = ({ onSearch }) => {
  const searchField = React.useRef(null)

  React.useEffect(() => {
    searchField.current.focus()
  }, [])

  const submitKeyword = event => {
    event.preventDefault()
    onSearch(searchField.current.value)
  }

  return (
    <form className='search-container' onSubmit={submitKeyword}>
      <input type='text' className='search-field' placeholder='Search all the GIFs' ref={searchField} />
      <button type='submit' className='search-button'>
        SEARCH
      </button>
    </form>
  )
}

export default SearchForm
