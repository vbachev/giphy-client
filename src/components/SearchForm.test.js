import React from 'react'
import renderer from 'react-test-renderer'
import SearchForm from './SearchForm'

test('it renders the search form', () => {
  const component = renderer.create(
    <SearchForm onSearch={() => {}} />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
