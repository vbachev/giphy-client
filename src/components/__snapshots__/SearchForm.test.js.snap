// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the search form 1`] = `
<form
  className="search-container"
  onSubmit={[Function]}
>
  <input
    className="search-field"
    placeholder="Search all the GIFs"
    type="text"
  />
  <button
    className="search-button"
    type="submit"
  >
    SEARCH
  </button>
</form>
`;
